flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/storage/SortBufferAccumulatorTest.java

```diff
@@ -0,0 +1,148 @@
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.apache.flink.runtime.io.network.partition.hybrid.tiered.TieredStorageTestUtils.generateRandomData;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/** Tests for {@link SortBufferAccumulator}. */
+class SortBufferAccumulatorTest {
+
+    private static final int NUM_TOTAL_BUFFERS = 1000;
+
+    private static final int BUFFER_SIZE_BYTES = 1024;
+
+    private static final float NUM_BUFFERS_TRIGGER_FLUSH_RATIO = 0.6f;
+
+    private NetworkBufferPool globalPool;
+
+    @BeforeEach
+    void before() {
+        globalPool = new NetworkBufferPool(NUM_TOTAL_BUFFERS, BUFFER_SIZE_BYTES);
+    }
+
+    @AfterEach
+    void after() {
+        globalPool.destroy();
+    }
+
+    @Test
+    void testAccumulateRecordsAndGenerateBuffers() throws IOException {
+        int numBuffers = 10;
+        int numRecords = 1000;
+        int indexEntrySize = 16;
+        TieredStorageSubpartitionId subpartitionId = new TieredStorageSubpartitionId(0);
+        Random random = new Random(1234);
+        TieredStorageMemoryManager tieredStorageMemoryManager =
+                createStorageMemoryManager(numBuffers);
+
+        int numExpectBytes = 0;
+        int numExpectBuffers = 0;
+        AtomicInteger numReceivedFinishedBuffer = new AtomicInteger(0);
+
+        try (SortBufferAccumulator bufferAccumulator =
+                new SortBufferAccumulator(1, 2, BUFFER_SIZE_BYTES, tieredStorageMemoryManager)) {
+            bufferAccumulator.setup(
+                    ((subpartition, buffers) ->
+                            buffers.forEach(
+                                    buffer -> {
+                                        numReceivedFinishedBuffer.incrementAndGet();
+                                        buffer.recycleBuffer();
+                                    })));
+            int currentBufferWriteBytes = 0;
+            boolean isBroadcastForPreviousRecord = false;
+            for (int i = 0; i < numRecords; i++) {
+                int numBytes = random.nextInt(BUFFER_SIZE_BYTES) + 1;
+                ByteBuffer record = generateRandomData(numBytes, random);
+                boolean isBroadcast = random.nextBoolean();
+                bufferAccumulator.receive(
+                        record, subpartitionId, Buffer.DataType.DATA_BUFFER, isBroadcast);
+                if (numExpectBytes + numBytes + indexEntrySize > BUFFER_SIZE_BYTES
+                        || i > 0 && isBroadcastForPreviousRecord != isBroadcast) {
+                    numExpectBuffers++;
+                    numExpectBytes = 0;
```

Comment:
This name is a bit strange, it feels like all the bytes expected, but in reality, it's not.

Reply:
Yes, renamed it to `currentBufferWrittenBytes`.